src/integrationtests/java/com/aws/greengrass/integrationtests/deviceauth/EvaluateClientDeviceActionsWithPolicyVariablesTest.java

```diff
@@ -0,0 +1,267 @@
+        startNucleusWithConfig(configFileName, State.RUNNING);
+    }
+
+    private void startNucleusWithConfig(String configFileName, State expectedServiceState) throws InterruptedException {
+        CountDownLatch authServiceRunning = new CountDownLatch(1);
+        kernel.parseArgs("-r", rootDir.toAbsolutePath().toString(), "-i",
+                getClass().getResource(configFileName).toString());
+        GlobalStateChangeListener listener = (GreengrassService service, State was, State newState) -> {
+            if (service.getName().equals(ClientDevicesAuthService.CLIENT_DEVICES_AUTH_SERVICE_NAME)
+                    && service.getState().equals(expectedServiceState)) {
+                authServiceRunning.countDown();
+            }
+        };
+        kernel.getContext().addGlobalStateChangeListener(listener);
+        kernel.launch();
+        assertThat(authServiceRunning.await(30L, TimeUnit.SECONDS), is(true));
+        kernel.getContext().removeGlobalStateChangeListener(listener);
+    }
+
+    private static void authzClientDeviceAction(GreengrassCoreIPCClient ipcClient,
+                                                AuthorizeClientDeviceActionRequest request,
+                                                Consumer<AuthorizeClientDeviceActionResponse> consumer)
+            throws Exception {
+        AuthorizeClientDeviceActionResponseHandler handler =
+                ipcClient.authorizeClientDeviceAction(request, Optional.empty());
+        AuthorizeClientDeviceActionResponse response = handler.getResponse().get(1, TimeUnit.SECONDS);
+        consumer.accept(response);
+    }
+
+    @AfterEach
+    void afterEach() {
+        LogConfig.getRootLogConfig().reset();
+        kernel.shutdown();
+    }
+
+    @Test
+    void GIVEN_cdaWithDeviceGroupPolicy_WHEN_AuthorizeClientDeviceValidResourceAction_THEN_returnTrue() throws Exception {
+
+        // make IPC call
+        try (EventStreamRPCConnection connection = IPCTestUtils.getEventStreamRpcConnection(kernel,
+                "BrokerWithAuthorizeClientDeviceActionPermission")) {
+            GreengrassCoreIPCClient ipcClient = new GreengrassCoreIPCClient(connection);
+
+            String deviceToken = getClientDeviceSessionAuthToken("MyThing", clientPem);
+
+            AuthorizeClientDeviceActionRequest connectRequest =
+                    new AuthorizeClientDeviceActionRequest().withOperation("mqtt:connect").withResource("mqtt:MyThing:foo")
+             .withClientDeviceAuthToken(deviceToken);
+
+            AuthorizeClientDeviceActionRequest publishRequest =
+                    new AuthorizeClientDeviceActionRequest().withOperation("mqtt:publish").withResource("mqtt:topic:MyThing")
+                            .withClientDeviceAuthToken(deviceToken);
+
+            Pair<CompletableFuture<Void>, Consumer<AuthorizeClientDeviceActionResponse>> cb =
+                    asyncAssertOnConsumer((m) -> {
```

Comment:
assertOnConsumer is typically used in callbacks where we need to validate the consumer is only called an expected number of times---can we simplify by just calling `assertTrue(m.isIsAuthorized());` directly in `authZClientDeviceAction`?